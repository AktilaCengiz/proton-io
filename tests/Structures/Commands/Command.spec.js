/* eslint-disable no-undef */
const Command = require("../../../src/Structures/Commands/Command");

test("Null props", () => {
    const NullProps = new Command("NullProps");
    expect(NullProps.id).toBe("NullProps");
    expect(NullProps.aliases).toBeNull();
    expect(NullProps.clientPermissions).toBeNull();
    expect(NullProps.cooldown).toBeNull();
    expect(NullProps.information).toBeNull();
    expect(NullProps.ownerOnly).toBe(false);
    expect(NullProps.userPermissions).toBeNull();
    expect(NullProps.executable).toBe(true);
    expect(NullProps).toMatchSnapshot();
});

test("Example", () => {
    const Example = new Command("Example", {
        aliases: ["example"],
        clientPermissions: [],
        cooldown: 10000,
        information: {},
        ownerOnly: true,
        userPermissions: []
    });
    expect(Example.id).toBe("Example");
    expect(Example.aliases).toBeInstanceOf(Array);
    expect(Example.clientPermissions).toStrictEqual([]);
    expect(Example.cooldown).toBe(10000);
    expect(Example.information).toStrictEqual({});
    expect(Example.ownerOnly).toBe(true);
    expect(Example.userPermissions).toStrictEqual([]);
    expect(Example.executable).toBe(true);
    expect(Example).toMatchSnapshot();
});
