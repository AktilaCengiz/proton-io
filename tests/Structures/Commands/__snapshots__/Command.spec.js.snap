// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example 1`] = `
Command {
  "advancedArgs": true,
  "aliases": Array [
    "example",
  ],
  "args": Array [],
  "category": "default",
  "client": null,
  "clientPermissions": Array [],
  "cooldown": 10000,
  "executable": true,
  "filepath": null,
  "handler": null,
  "id": "Example",
  "information": Object {},
  "ownerOnly": true,
  "rateLimit": null,
  "typing": null,
  "userPermissions": Array [],
}
`;

exports[`Null props 1`] = `
Command {
  "advancedArgs": null,
  "aliases": null,
  "args": null,
  "category": "default",
  "client": null,
  "clientPermissions": null,
  "cooldown": null,
  "executable": true,
  "filepath": null,
  "handler": null,
  "id": "NullProps",
  "information": null,
  "ownerOnly": false,
  "rateLimit": null,
  "typing": null,
  "userPermissions": null,
}
`;
